lock/src/test/java/org/infinispan/lock/api/ClusteredLockSplitBrainTest.java

```diff
@@ -0,0 +1,140 @@
+         expectException(AvailabilityException.class, () -> clm.defineLock("lock"));
+      }
+   }
+
+   @Test
+   public void testLockUseAfterPartitionWithoutMajority() throws Throwable {
+
+      ClusteredLockManager clm0 = ClusteredLockManagerFactory.from(getCacheManagers().get(0));
+      ClusteredLockManager clm1 = ClusteredLockManagerFactory.from(getCacheManagers().get(1));
+      ClusteredLockManager clm2 = ClusteredLockManagerFactory.from(getCacheManagers().get(2));
+      ClusteredLockManager clm3 = ClusteredLockManagerFactory.from(getCacheManagers().get(3));
+
+      clm0.defineLock(LOCK_NAME);
+
+      assertTrue(clm0.isDefined(LOCK_NAME));
+      ClusteredLock lock0 = clm0.get(LOCK_NAME);
+      assertNotNull(lock0);
+
+      ClusteredLock lock1 = clm1.get(LOCK_NAME);
+      assertNotNull(lock1);
+
+      ClusteredLock lock2 = clm2.get(LOCK_NAME);
+      assertNotNull(lock2);
+
+      ClusteredLock lock3 = clm3.get(LOCK_NAME);
+      assertNotNull(lock3);
+
+      splitCluster(new int[]{0, 1}, new int[]{2, 3});
+
+      await(lock0.lock().exceptionally(ex -> {
+         assertException(ClusteredLockException.class, ex);
+         assertException(AvailabilityException.class, ex.getCause());
+         return null;
+      }));
+      await(lock1.lock().exceptionally(ex -> {
+         assertException(ClusteredLockException.class, ex);
+         assertException(AvailabilityException.class, ex.getCause());
+         return null;
+      }));
+      await(lock2.lock().exceptionally(ex -> {
+         assertException(ClusteredLockException.class, ex);
+         assertException(AvailabilityException.class, ex.getCause());
+         return null;
+      }));
+      await(lock3.lock().exceptionally(ex -> {
+         assertException(ClusteredLockException.class, ex);
+         assertException(AvailabilityException.class, ex.getCause());
+         return null;
+      }));
+   }
+
+   @Test
+   public void testLockUseAfterWithMajorityPartition() throws Throwable {
+
+      ClusteredLockManager clm0 = ClusteredLockManagerFactory.from(getCacheManagers().get(0));
+      ClusteredLockManager clm1 = ClusteredLockManagerFactory.from(getCacheManagers().get(1));
+      ClusteredLockManager clm2 = ClusteredLockManagerFactory.from(getCacheManagers().get(2));
+      ClusteredLockManager clm3 = ClusteredLockManagerFactory.from(getCacheManagers().get(3));
+
+      assertTrue(clm0.defineLock(LOCK_NAME));
+      assertFalse(clm1.defineLock(LOCK_NAME));
+      assertFalse(clm2.defineLock(LOCK_NAME));
+      assertFalse(clm3.defineLock(LOCK_NAME));
+
+      ClusteredLock lock0 = clm0.get(LOCK_NAME);
+      ClusteredLock lock1 = clm1.get(LOCK_NAME);
+      ClusteredLock lock2 = clm2.get(LOCK_NAME);
+      ClusteredLock lock3 = clm3.get(LOCK_NAME);
+
+      splitCluster(new int[]{0, 1, 2}, new int[]{3});
+
+      await(lock0.lock()
+            .thenRun(() -> {
+               assertTrue("we locked with node A", true);
```

Comment:
weird `assertTrue(true)`

Reply:
This is just a way to document, but I can change it

Reply:
I will do something smarter